ZenPacks/zenoss/LinuxMonitor/HardDisk.py

```diff
@@ -0,0 +1,39 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2016, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+from Products.ZenModel.HardDisk import HardDisk as BaseHardDisk
+from . import zenpacklib
+
+
+class HardDisk(BaseHardDisk):
+    meta_type = 'LinuxHardDisk'
+
+    _properties = BaseHardDisk._properties + (
+        {'id': 'physicalvolumes', 'label': 'Physical Volumes',
+            'type': 'entity', 'mode': 'w'},
+        {'id': 'size', 'label': 'Size',
+            'type': 'string', 'mode': 'w'}
+        )
+
+    def getPhysicalVolumes(self):
+        results = zenpacklib.catalog_search(
+            self.device(),
+            'PhysicalVolume',
+            harddisk_id=self.id)
+        physicalvolumes = []
+        for result in results:
+            try:
+                physicalvolumes.append(result.getObject())
+            except Exception:
+                pass
+        self.physicalvolumes = physicalvolumes
```

Comment:
You shouldn't be assigning a list of persistent objects as a property of the HardDisk persistent object. This self.physicalvolumes assignment needs to go away, as does its entry in _properties above.